src/org/thoughtcrime/securesms/ShareActivity.java

```diff
@@ -213,7 +213,10 @@ private void handleResolvedMedia(Intent intent) {
 
   private Intent getBaseShareIntent(final @NonNull Class<?> target) {
     final Intent intent = new Intent(this, target);
-    setSharedText(intent, getIntent().getStringExtra(Intent.EXTRA_TEXT));
+    CharSequence text = getIntent().getCharSequenceExtra(Intent.EXTRA_TEXT);
```

Comment:
after all this mess, i would be a bit more conservative and call getCharSequenceExtra() only if getStringExtra() returned null before.

i would not be surprised if there are implementation that (willI) return null on getCharSequenceExtra() and the text on getStringExtra().